### ISSUE TITLE
Feature to get a word from its vector and vice-versa

### ISSUE THREAD

**Issue by Vijethbv** [maciejkula/glove-python#51]
I'd like to get a word vector from its associated word and vice-versa.
Is it possible to do this in existing version of the code?

**Comment by draperunner** [comment]
Yes, you can. If I am not entirely mistaken, the word ids contained in the `Glove` class' `self.dictionary` correspond to the word's vector's index in `self.word_vectors`.

I did this to save every word and its vector in a text file (with first line being number of vectors and dimension of vectors):

``` python
# Sort words by their id
word_and_word_id_tuples = list(glove_model.dictionary.items())
word_and_word_id_tuples.sort(key=lambda t: t[1])  # id is second element of tuple

# Save vectors to output file
with open(args.out_file, 'w+') as f:
    f.write(str(len(word_and_word_id_tuples)) + " " + str(args.dimension) + "\n")
    for word, word_id in word_and_word_id_tuples:
        f.write(word + " " + " ".join([str(value) for value in glove_model.word_vectors[word_id]]) + "\n")
```

**Comment by Vijethbv** [comment]
Thank you for your response.

**Comment by ibarrien** [comment]
Here is how to make a dictionary with keys = words the model was trained on and values = glove representation of the words, *in one line*

Suppose model = fitted Glove() model, then simply define the desired dictionary as:
`D = {word: model.word_vectors[model.dictionary[word]] for word in model.dictionary.keys()}`

Example: suppose "music" was in the training corpus and model.no_components = 200. Then `D["music"]` is a vector of floats of length 200.

You can then save this dictionary using pickle (serialization) or write it as a text file in your preferred format.
